internal/services/mssqlmanagedinstance/mssql_managed_instance_start_stop_schedule_resource.go

```diff
@@ -0,0 +1,352 @@
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			scheduleID := parse.NewManagedInstanceStartStopScheduleID(id.SubscriptionId, id.ResourceGroupName, id.ManagedInstanceName, "default")
+			metadata.SetID(scheduleID)
+
+			return nil
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			managedInstanceId, err := commonids.ParseSqlManagedInstanceID(metadata.ResourceData.Get("managed_instance_id").(string))
+			if err != nil {
+				return err
+			}
+
+			var model SqlManagedInstanceStartStopScheduleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *managedInstanceId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *managedInstanceId, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", managedInstanceId)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Properties.Description = &model.Description
+				} else {
+					properties.Properties.Description = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("schedule_list") {
+				properties.Properties.ScheduleList = pointer.From(expandScheduleItemModelArray(model.ScheduleList))
+			}
+
+			if metadata.ResourceData.HasChange("timezone_id") {
+				if model.TimeZoneId != "" {
+					properties.Properties.TimeZoneId = &model.TimeZoneId
+				} else {
+					properties.Properties.TimeZoneId = nil
+				}
+			}
+
+			properties.SystemData = nil
+
+			if _, err := client.CreateOrUpdate(ctx, *managedInstanceId, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *managedInstanceId, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			managedInstanceId, err := commonids.ParseSqlManagedInstanceID(metadata.ResourceData.Get("managed_instance_id").(string))
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *managedInstanceId)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(managedInstanceId)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *managedInstanceId, err)
+			}
+
+			state := SqlManagedInstanceStartStopScheduleModel{
+				SqlManagedInstanceId: managedInstanceId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if name := model.Name; name != nil {
```

Comment:
Use the `name` from the parsed ID instead of from the response, because the upstream API might have unexpected changes.

Reply:
Updated